components/empty-state/empty-state-action-button.js

```diff
@@ -0,0 +1,101 @@
+import '../button/button.js';
+import '../button/button-subtle.js';
+import { css, html, LitElement, nothing } from 'lit';
+
+/**
+ * `d2l-empty-state-action-button` is an empty state action component that can be placed inside of the default slot of `empty-state-simple` or `empty-state-illustrated` to add a button action to the component.
+ * @fires d2l-empty-state-action - Dispatched when the action button is clicked
+ */
+class EmptyStateActionButton extends LitElement {
+
+	static get properties() {
+		return {
+			/**
+			 * REQUIRED: The action text to be used in the button
+			 * @type {string}
+			 */
+			text: { type: String },
+			/**
+			 * This will change the action button to use a primary button instead of the default subtle button. The primary attribute is only valid when `d2l-empty-state-action-button` is placed within `d2l-empty-state-illustrated` components
+			 * @type {boolean}
+			 */
+			primary: { type: Boolean },
+			_illustrated: { state: true }
+		};
+	}
+
+	static get styles() {
+		return css`
+					.d2l-empty-state-action {
+						vertical-align: top;
+					}
+				`;
+	}
+
+	constructor() {
+		super();
+		this._missingTextErrorHasBeenThrown = false;
+		this._validatingTextTimeout = null;
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		requestAnimationFrame(() => {
+			const e = new CustomEvent('d2l-empty-state-illustrated-check', {
+				bubbles: true,
+				detail: {}
+			});
+			this.dispatchEvent(e);
+			this._illustrated = e.detail.illustrated;
+		});
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+		this._validateText();
+	}
+
+	render() {
+		let actionButton = nothing;
+		if (this.text) {
+			actionButton = this._illustrated && this.primary
+				? html`<d2l-button
+							class="d2l-empty-state-action"
+							@click=${this._handleActionClick}
+							primary>
+								${this.text}
+						</d2l-button>`
+				: html`<d2l-button-subtle
+							class="d2l-empty-state-action"
+							@click=${this._handleActionClick}
+							?slim=${!this._illustrated}
+							text=${this.text}>
+						</d2l-button-subtle>`;
+		}
+		return html`${actionButton}`;
+	}
+
+	_handleActionClick(e) {
+		e.stopPropagation();
+		this.dispatchEvent(new CustomEvent('d2l-empty-state-action', {
+			bubbles: true
```

Comment:
Is bubbling actually required? I'd expect almost all consumers to wire up directly to the action.

Reply:
You're right, that makes more sense. I wrote the documentation that way but when I was writing this I was thinking that consumers would be listening for events on the container empty state component.